lock_api/src/lib.rs

```diff
@@ -47,7 +47,13 @@
 //!     }
 //!
 //!     fn try_lock(&self) -> bool {
-//!         self.0.swap(true, Ordering::Acquire)
+//!         match self
+//!             .0
+//!             .compare_exchange_weak(false, true, Ordering::Acquire, Ordering::Relaxed)
+//!         {
+//!             Ok(_) => true,
+//!             Err(_) => false,
+//!         }
```

Comment:
You want a strong cmpxchg for `try_lock`, to avoid spurious failures. Also just use `.is_ok()` instead of a match.